### ISSUE TITLE
[Bug]: Outdated node IDs in DevTools when refreshing a page containing iframes

### ISSUE THREAD

**Issue by flotwig** [electron/electron#32423]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

16.0.6

### What operating system are you using?

Ubuntu, macOS

### Operating System Version

* `Linux hostname 5.15.11-76051511-generic #202112220937~1640185481~21.10~b3a2c21 SMP Wed Dec 22 15:41:49 U x86_64 x86_64 x86_64 GNU/Linux`
* macOS Catalina, Big Sur

### What arch are you using?

x64

### Last Known Working Electron version

Broken since at least 7.1.10 (Cypress 3.8.3)

### Expected Behavior

Chrome DevTools works between page reloads.

### Actual Behavior

When running a Cypress test inside of Electron, reloading the test runner BrowserWindow breaks the DevTools interface. The element tree breaks until you close and reopen DevTools.

Video: 

https://user-images.githubusercontent.com/1151760/145282471-70f64095-99e5-47dc-a3ff-bd5977e796f8.mp4

### Additional Information

There are notes from my investigation on the Cypress issue: https://github.com/cypress-io/cypress/issues/15932#issuecomment-989207945

TL;DR:

* Running the same repro steps inside of Chrome 96 instead of Electron does not reproduce the issue, leading me to believe this is an Electron-specific issue.
* If you use `--remote-debugging-port=12345` with Electron and visit `http://localhost:12345` to debug, the issue still occurs when using external DevTools.
* It seems to have to do something with asynchronously-created `iframe`s. If I remove the `iframe` creation from Cypress, DevTools works properly. I attempted to create a stand-alone repro here but it did not reproduce the issue: https://github.com/flotwig/electron-devtools-issue-repro
* Using the protocol monitor, you can see `{code: -32000, message: "Could not find node with given id"}` errors when hovering over the empty elements, leading me to believe this is some kind of bug with the internal state of the DevTools client: ![image](https://user-images.githubusercontent.com/1151760/148998188-4557d9c1-12c9-49c5-b5ee-c1966bc7a9ee.png)

---

I understand that the repro might be a little too involved at this point. I could use additional guidance on what differences could exist between DevTools in Electron vs. vanilla Chrome, to help debug this further.
